code/src/Components/TaskInput.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react"
+import { useDispatch } from "react-redux"
+
+import { todo } from "../Reducers/todo"
+import { v4 as uuidv4 } from 'uuid' // for unique id
+
+export const TaskInput = () => {
+  const [input, setInput] = useState("")
+  const dispatch = useDispatch()
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+      dispatch(todo.actions.addItem({
+      newItem: {
+        id: uuidv4(), 
+        text: input,
+        isComplete: false,
+      }
+    }))
+    setInput("")
+  }
+  
+  return (
+    <form onSubmit={handleSubmit}>
+      <label className="task-input">
+        <input
+          className="task-input-field"
+          placeholder="Add a new task..."
+          type="text"
+          value={input}
+          onChange={(event) => setInput(event.target.value)}
+        />
+      </label>
+      <button className="task-add-button" type="submit" onClick={handleSubmit}>
```

Comment:
This shouldn't be need 

`onClick={handleSubmit}`

```suggestion
      <button className="task-add-button" type="submit">
```

since you on line 24 say the to the form `onSubmit={handleSubmit}`

Try to remove and see if it still works!

Reply:
I tried it out and it works :D Thank you!!